src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickingBlocks.java

```diff
@@ -0,0 +1,165 @@
+package io.github.thebusybiscuit.slimefun4.implementation.tasks;
+
+import io.github.thebusybiscuit.cscorelib2.blocks.BlockPosition;
+import org.bukkit.Location;
+import org.bukkit.World;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+/**
+ * Memory efficient, synchronized collection of {@link BlockPosition BlockPositions} used by the
+ * {@link TickerTask}. All methods are thread-safe unless explicitly stated otherwise.
+ *
+ * @author md5sha256
+ */
+final class TickingBlocks {
+
+    private final Map<World, Set<Long>> blocks = Collections.synchronizedMap(new WeakHashMap<>());
+
+    /**
+     * Get the compressed {@link BlockPosition BlockPositions} mapped to a given {@link World}.
+     * The returned {@link Set} is thread-safe. Changes to set will be reflected in this instance.
+     *
+     * @param world The world
+     * @return Returns a {@link Set} of {@link Long Longs} which are compressed {@link BlockPosition BlockPositions}
+     *
+     * @see #getBlocksCopy()
+     */
+    public @Nonnull Set<Long> getBlocks(@Nonnull World world) {
+        return blocks.computeIfAbsent(world, (unused) -> Collections.synchronizedSet(new HashSet<>()));
```

Comment:
Don't put braces around a single variable in a lambda expression
```suggestion
        return blocks.computeIfAbsent(world, x -> Collections.synchronizedSet(new HashSet<>()));
```